mi/dataset/driver/zplsc_c/zplsc_echogram_generator.py

```diff
@@ -0,0 +1,633 @@
+
+                # If in process mode, get the latest date that has 24 1-hour data files for echogram generation.
+                if self.process_mode:
+                    echogram_dates[self.get_latest_echogram_date(date_dirs_url, date_dirs_list)] = False
+
+                # Otherwise, get all the year/month date subdirectories for this subsite and deployment.
+                else:
+                    for date_dir in date_dirs_list:
+                        year = int(date_dir[0])
+                        month = int(date_dir[1])
+
+                        # Save the date and indicate that the entire month should be generated.
+                        echogram_dates[date(year, month, 1)] = True
+
+        return echogram_dates, date_dirs_url
+
+    def get_deployment_dirs(self, subsite):
+        """
+        This method will determine the deployment directories for the subsite
+        passed in.
+
+        :param subsite: The subsite of the ZPLSC instrument.
+        :return: deployments: The list of deployment directories.
+        """
+
+        deployments = [deployment for deployment in self.deployments]
+        if not deployments:
+            # Generate the portion of the URL up to the subsite directory.
+            subsite_url = join(RAW_DATA_URL, subsite.upper())
+            try:
+                # Get the all the deployment sub-directories under the subsite directory.
+                deployment_dirs = self.get_dir_contents(subsite_url)
+            except urllib2.HTTPError:
+                raise
+
+            # Generate the list of the 24 1-hour raw data files for the given date.
+            deployment_list = DEPLOYMENT_DIR_MATCHER.findall(deployment_dirs)
+            deployment_list.sort()
+
+            if self.process_mode:
+                deployment_list = [deployment_list[-1]]
+
+            for deployment in deployment_list:
+                try:
+                    deployments.append(int(deployment))
+
+                except ValueError as ex:
+                    log.error('Invalid deployment number: %s: %s', deployment, ex.message)
+                    break
+
+        return deployments
+
+    def store_echogram(self, zplsc_echogram_file):
+        """
+        This method will store the generated echogram in the location specified
+        in the configuration file. It will then clean up the temporary folder.
+
+        :param zplsc_echogram_file: The generated ZPLSC echogram
+        :return:
+        """
+
+        # Create the echogram directory, then copy the echogram to the local directory.
+        self.echogram_directory = expanduser(self.echogram_directory)
+        if not exists(self.echogram_directory):
+            mkdir(self.echogram_directory)
+
+        system('cp ' + zplsc_echogram_file + ' ' + self.echogram_directory)
+
+        # Clean up the temporary files.
+        system('rm -f ' + join(self.temp_directory, '*'))
```

Comment:
i think there's a way to do this without doing a system call.

Reply:
Replace system call with os.remove call with a loop.